### ISSUE TITLE
Add Zitadel or Generic OIDC provider

### ISSUE THREAD

**Issue by s0l4r** [edgedb/edgedb#6908]
It would be very nice to have support for Zitadel as an IdP in EdgeDB auth:
https://github.com/zitadel/zitadel

https://zitadel.com/docs/apis/openidoauth/endpoints
https://zitadel.com/docs/guides/integrate/oauth-recommended-flows

If it's not prioritized, maybe you can implement a generic OIDC provider to start with.

Thanks!

Here are example values needed for a generic OIDC provider:
![generic-oidc](https://github.com/edgedb/edgedb/assets/2366752/8b6b0029-6b00-4c18-9189-91c5c0e6006d)

![generic-oidc-optional](https://github.com/edgedb/edgedb/assets/2366752/d4dc91b9-f77d-41ce-88de-6da18ad83db3)

**Comment by scotttrinh** [comment]
Yeah, I think we will support having generic OIDC providers at some point, but we don't have a clear place on the roadmap for it just yet. It's something we talked about from the beginning and designed the schema to have a concept of "builtin" providers vs custom ones. We'll keep this issue open and use it for further discussion and to gather community feedback, thanks for starting the conversation!

**Comment by scotttrinh** [comment]
@s0l4r 
Implementing this for the upcoming 6.0 release. Available for early testing via nightly! #7510
